rmw_iceoryx_cpp/test/iceoryx_name_conversion_test.cpp

```diff
@@ -0,0 +1,63 @@
+// Copyright (c) 2020 by Robert Bosch GmbH. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "rmw_iceoryx_cpp/iceoryx_name_conversion.hpp"
+
+#include <gtest/gtest.h>
+
+TEST(NameConverisonTests, get_name_n_type_from_iceoryx_service_description)
+{
+  auto topic_and_type = rmw_iceoryx_cpp::get_name_n_type_from_iceoryx_service_description(
+    "SERVICE",
+    "INSTANCE",
+    "EVENT");
+  EXPECT_EQ(std::get<0>(topic_and_type), "/INSTANCE/SERVICE/EVENT");
+  EXPECT_EQ(std::get<1>(topic_and_type), "service_ara_msgs/msg/EVENT");
+}
+
+TEST(NameConverisonTests, get_service_description_elements_revers)
+{
+  auto service_description = rmw_iceoryx_cpp::get_service_description_elements(
```

Comment:
why does this API have to be public? Can't we just test `get_name_n_type_from_iceoryx_service_description` and validate the string with `std::get<0>`?

Reply:
I guess you mean `get_iceoryx_service_description`. The issue is that there is a type support required and I had no idea, where to get it from. I also did not want to start mocking it. Actually I am not sure how to mock a single function in the common cpp file.